### PR TITLE
PSEC_1655_Set_up_Query_Log_centralisation

### PR DESCRIPTION
**Pull request by wunzeco**

### CODE REVIEW

File: src/aws_scanner_argument_parser.py [hunk 1]
```diff
@@ -261,6 +261,7 @@ def _add_audit_route53_query_logs(self, subparsers: Any) -> None:
         self._add_accounts_args(audit_parser)
         self._add_verbosity_arg(audit_parser)
         self._add_enforce_arg(audit_parser, "add centralised query logs to Route53 Zones that don't already have one")
+        audit_parser.add_argument("-w", "--with_subscription_filter", type=bool, help="create subscription filter")
```
Comment (cob16) [inline src/aws_scanner_argument_parser.py:264]: this line looks common in 3

Might be time to extract to a helper method

File: src/clients/composite/aws_route53_client.py [hunk 2]
```diff
@@ -91,24 +93,55 @@ def _create_query_log_actions(
...
+            if (
+                self._logs.is_central_log_group(log_group=log_group, service_name=ServiceName.route53)
+                and not with_subscription_filter
+            ):
+                actions.append(
+                    DeleteLogGroupSubscriptionFilterAction(
+                        logs=self._logs, config=self._config, service_name=ServiceName.route53
+                    )
+                )
+            if (
```
Comment (cob16) [inline src/clients/composite/aws_route53_client.py:109]:
```suggestion
            elif (
```
Could be a elif

**Review by cob16 — APPROVED** [review]

File: src/clients/composite/aws_route53_client.py (hunk 2 above)
Comment (aminebz949904) [reply src/clients/composite/aws_route53_client.py:109]: done

File: src/aws_scanner_argument_parser.py (hunk 1 above)
Comment (aminebz949904) [reply src/aws_scanner_argument_parser.py:264]: refactored accordingly

**Review by cob16 — APPROVED** [review]

**Pull request merged**